Tuturial-Structure-Template.md

```diff
@@ -1,68 +1,178 @@
+> - Dev.to or any markdown file - within code blocks:
+> 
+        > ```javascript 
+        > code blocks 
+        > Always remember to put the coding language 
+        > ```
+    >
+> For example:
+
+To add step-up-authentication to your Agora client:
+
+1. Do this.
+1. Do that:
+   1. Sometimes you need substeps.
+   1. Always at least 2 substeps.  
+1. Do something else.
+
+### Build \<module2 in procedure>
+
+> Brief explanation about what the technology does in this section, then a procedure.
+> 
+> For example:
+
+To add step-up-authentication to your security infrastructure:
+
+1. Do this.
+1. Do that:
+    1. Sometimes you need substeps.
+    1. Always at least 2 substeps.
+1. Do something else.
+
+
+## Test \<A rewording of the page title>
+
+To ensure that you have implemented \<A rewording of the page title> correctly:
+
+1. Do this.
+1. Do that:
+    1. Sometimes you need substeps.
+    1. Always at least 2 substeps.
+1. Do something else.
+
+## Conclusion
 
-## Conclusion ##
 Write a short paragraph in conclusion. At the end of the paragraph, please include a link to the completed source code on GitHub.  
 
-## Other Resources ## 
-> (use as section title)
+> Always include a section at the end of each post that includes links to the relevant Agora Documentation and Advanced Guides.
+
+Your next steps after implementing this functionality are:
+
+1. Read the following documentation:
+    1. Very good document about something useful.
+   1. Another very good document about something useful.
+1. Join the Agora.io Developer Slack community.
+1. Something else. 
+
+## Reference
+
+> If there is reference information you want to share with the user, add it here. If not, remove this section. 
+> 
+> For example:
+
+You use the following to configure or develop \<TITLE>:
+
+| Name        | Default           | Type  | Description |
+| ------------- |:-------------:| -----:| :-----|
+| shakespeare     | playwright | job | Set `shakespeare` to `poet` to enable the sonnet writing functionality for this instance.  Possible values are: <ul><li>`actor` - someone who pretends to be someone else while performing in a film, play, or television or radio programme.</li><li>`director` - a person who is in charge of a film or play and tells the actors how to play their parts.</li><li> `impresario` - a person who arranges different types of public entertainment, such as theatre, musical, and dance events.</li><li> `playwright` - a person who writes plays.</li><li> `poet` - a person who writes poems.</li></uls>
+|You| get|the | idea |
+
+
```

Comment:
Other information such as FAQs, do they go to the conclusion section or the reference section?

Reply:
I understand that the Conclusion section only contains Advanced Guides that users might need after completing this guide? So FAQs should be in reference?

Reply:
I hate FAQs. It means the doc is not written well enough. 

Conclusion just points out what the end user has learned. 

Thanks